component/monitoring.jsonnet

```
@@ -31,6 +31,28 @@ local service_monitor = com.namespaced(params.namespace, {
   },
 });
 
+local asciiTitleCase(str) = std.asciiUpper(str[0]) + str[1:];
+
+local failed_job_types = [ 'archive', 'backup', 'check', 'prune', 'restore' ];
+local render_failed_job_alert(type) =
+  assert std.member(failed_job_types, type) : 'Unknown failed job type "%s"' % type;
+  local alertconfig = params.job_failed_alerts_for[type];
+  if alertconfig.enabled then
+    params.job_failed_alert_template + alertconfig.overrides + {
+      alert: super.alert % { type: asciiTitleCase(type) },
+      expr: super.expr % { type: type },
+    }
```

Probably a good idea to allow merging of configs, e.g. to add labels or annotations on individual alerts:

```suggestion
    com.makeMergeable(params.job_failed_alert_template) + 
    com.makeMergeable(alertconfig.overrides) + 
    {
      alert: super.alert % { type: asciiTitleCase(type) },
      expr: super.expr % { type: type },
    }
```

With the current setup, setting `overrides.labels` for an alert would replace the existing labels coming from the alert template.

---

🧙‍♂️